contracts/locked/LockedSOV.sol

```diff
@@ -0,0 +1,371 @@
+
+		emit RegistryCliffAndDurationUpdated(msg.sender, _vestingRegistry, _cliff, _duration);
+	}
+
+	/**
+	 * @notice Adds SOV to the user balance (Locked and Unlocked Balance based on `_basisPoint`).
+	 * @param _userAddress The user whose locked balance has to be updated with `_sovAmount`.
+	 * @param _sovAmount The amount of SOV to be added to the locked and/or unlocked balance.
+	 * @param _basisPoint The % (in Basis Point)which determines how much will be unlocked immediately.
+	 */
+	function deposit(
+		address _userAddress,
+		uint256 _sovAmount,
+		uint256 _basisPoint
+	) external {
+		bool txStatus = SOV.transferFrom(msg.sender, address(this), _sovAmount);
+		require(txStatus, "Token transfer was not successful. Check receiver address.");
+
+		uint256 unlockedBal = _sovAmount.mul(_basisPoint).div(10000);
+
+		unlockedBalances[_userAddress] = unlockedBalances[_userAddress].add(unlockedBal);
+		lockedBalances[_userAddress] = lockedBalances[_userAddress].add(_sovAmount).sub(unlockedBal);
+
+		emit Deposited(msg.sender, _userAddress, _sovAmount, _basisPoint);
+	}
+
+	/**
+	 * @notice Adds SOV to the locked balance of a user.
+	 * @param _userAddress The user whose locked balance has to be updated with _sovAmount.
+	 * @param _sovAmount The amount of SOV to be added to the locked balance.
+	 * @dev This is here because there are dependency with other contracts.
+	 */
+	function depositSOV(address _userAddress, uint256 _sovAmount) external {
+		bool txStatus = SOV.transferFrom(msg.sender, address(this), _sovAmount);
+		require(txStatus, "Token transfer was not successful. Check receiver address.");
+
+		lockedBalances[_userAddress] = lockedBalances[_userAddress].add(_sovAmount);
+
+		emit Deposited(msg.sender, _userAddress, _sovAmount, 0);
+	}
+
+	/**
+	 * @notice A function to withdraw the unlocked balance.
+	 * @param _receiverAddress If specified, the unlocked balance will go to this address, else to msg.sender.
+	 */
+	function withdraw(address _receiverAddress) external {
+		address userAddr = _receiverAddress;
+		if (_receiverAddress == address(0)) {
+			userAddr = msg.sender;
+		}
+
+		uint256 amount = unlockedBalances[msg.sender];
+		unlockedBalances[msg.sender] = 0;
+
+		bool txStatus = SOV.transfer(userAddr, amount);
+		require(txStatus, "Token transfer was not successful. Check receiver address.");
+
+		emit Withdrawn(msg.sender, userAddr, amount);
+	}
+
+	/**
+	 * @notice Creates vesting contract (if it hasn't been created yet) for the calling user.
+	 * @return _vestingAddress The New Vesting Contract Created.
+	 */
+	function createVesting() public returns (address _vestingAddress) {
+		_vestingAddress = _createVesting(msg.sender);
+
+		emit VestingCreated(msg.sender, _vestingAddress);
+	}
+
+	/**
+	 * @notice Stakes tokens for a user who already have a vesting created.
+	 * @dev The user should already have a vesting created, else this function will throw error.
+	 */
+	function stakeTokens() external {
```

Comment:
I would find it nice if there would be a function which 
1. withdraws the unlocked tokens and
2. stakes the locked tokens
at once

Reply:
Added in a57eae8